core/src/main/java/forklift/consumer/Consumer.java

```diff
@@ -483,10 +405,154 @@ else if (producer.topic().length() > 0)
+                }
+                String key = ((Config)decorator).field();
+                if (key.equals("")) {
+                    key = mappedName;
+                }
+                value = config.get(key);
+            }
+        } else if (decorator.annotationType() == Headers.class) {
+            final Map<Header, Object> headers = msg.getHeaders();
+            if (mappedClass == Map.class) {
+                value = headers;
+            } else {
+                final Header key = ((Headers)decorator).value();
+                if (headers == null) {
+                    log.warn("Attempt to inject {} from headers, but headers are null", key);
+                } else if (!key.getHeaderType().equals(mappedClass)) {
+                    log.warn("Injecting field {} failed because it is not type {}", mappedName, key.getHeaderType());
+                } else {
+                    value = headers.get(key);
+                }
+            }
+        } else if (decorator.annotationType() == forklift.decorators.Properties.class) {
+            Map<String, String> properties = msg.getProperties();
+            if (mappedClass == Map.class) {
+                value = msg.getProperties();
+            } else if (properties != null) {
+                String key = ((forklift.decorators.Properties)decorator).value();
+                if (key.equals("")) {
+                    key = mappedName;
+                }
+                if (properties == null) {
+                    log.warn("Attempt to inject field {} from properties, but properties is null", key);
+                } else {
+                    value = properties.get(key);
+                }
+            }
+        } else if (decorator.annotationType() == forklift.decorators.Producer.class) {
+            if (mappedClass == ForkliftProducerI.class) {
+                forklift.decorators.Producer producer = (forklift.decorators.Producer)decorator;
+                final ForkliftProducerI p;
+                if (producer.queue().length() > 0)
+                    p = forklift.getConnector().getQueueProducer(producer.queue());
+                else if (producer.topic().length() > 0)
+                    p = forklift.getConnector().getTopicProducer(producer.topic());
+                else
+                    p = null;
+                value = p;
+            }
+        }
+        return value;
+    }
+
     public Class<?> getMsgHandler() {
         return msgHandler;
     }
 
+    /**
+     * Creates an instance of hte MessageHandler class utilized by this constructor.  Constructor and Field level injection is performed using both the
+     * passed in msg and any Services {@link #addServices(ConsumerService...) added} to this consumer.
+     *
+     * @param msg
+     */
+    public Object getMsgHandlerInstance(ForkliftMessage msg) throws InvocationTargetException, IOException, InstantiationException, IllegalAccessException {
```

Comment:
I can't imagine most callers to these methods having a meaningful way of handling all of these exceptions differently. Maybe just wrap them in one exception?